docs/root/configuration/observability/access_log/usage.rst

```diff
@@ -310,6 +310,7 @@ The following command operators are supported:
       :ref:`strictly-checked header <envoy_v3_api_field_extensions.filters.http.router.v3.Router.strict_check_headers>` in addition to 400 response code.
     * **SI**: Stream idle timeout in addition to 408 response code.
     * **DPE**: The downstream request had an HTTP protocol error.
+    * **UPE**: The upstream response had an HTTP protocol error.
```

Comment:
I want to avoid one point of confusion and I have no ideas how to other than what you have here.

on the downstream side we can only really track protocol errors on the request side - if there's a bad request Envoy will 400.  If Envoy sends an invalid response, we either get a stream reset or disconnect, but no data.

On the upstream side, we can track both protocol errors on the posted request (if upstream responds with a 4xx) and on the posted response.  

I agree that for pure response codes, 503 applies to the bad response case, but I'm mildly concerned folks will conflate "bad request" and "bad response" with UPE since both are upstream-related protocol errors.

Unfortunately URPE isn't any more clear, and you do say exactly what's going on, so unless you can think of something more clever than I think this can stand :-P

Reply:
The logic detects protocol errors only in responses from upstream. In case Envoy sends a request with protocol error and upstream does not like it, the downstream will still receive 503.
But, as you said, I think description of UPE flag in access log clearly says "response".